Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -0,0 +1,152 @@
+            return 1
+        # Make sure we have the commit that user want to revert
+        try:
+            commits = map(lambda commit_arg: repository.find(commit_arg, include_log=False), args.commits)
+        except (local.Scm.Exception, ValueError) as exception:
+            # ValueErrors and Scm exceptions usually contain enough information to be displayed
+            # to the user as an error
+            sys.stderr.write(str(exception) + '\n')
+            return 1
+
+        commit_hashes = list(map(lambda commit: commit.hash, commits))
+        result = run([repository.executable(), 'revert', '--no-commit'] + commit_hashes, cwd=repository.root_path, capture_output=True)
+        if result.returncode:
+            # git revert wiil output nothing if this commit is already reverted
+            if not result.stdout.strip():
+                sys.stderr.write('The commits you spiecfied seems already be reverted.')
+                return 2
+            # print out
+            sys.stderr.write(result.stdout)
+            log.info('If you have merge conflicts, after resolving them, please use git-webkit pfr to publish your pull request')
+            return 1
+        # restore all ChangeLog changes, we should not revert those entries
+        result = run(
+            [repository.executable(), 'diff', '--name-status', '--staged'],
+            capture_output=True, encoding='utf-8', cwd=repository.root_path,
+        )
+        if result.returncode:
+            sys.stderr.write('Failed to detect any diff after revert.')
+            return 1
+        for line in result.stdout.splitlines():
+            state, file = line.split(None, 1)
+            if 'ChangeLog' in file:
+                result = run([repository.executable(), 'restore', '--staged', file], cwd=repository.root_path)
+                if result.returncode:
+                    sys.stderr.write('Failed to restore staged file: {}'.format(file))
+                    run([repository.executable(), 'revert', '--abort'], cwd=repository.root_path)
+                    return 1
+
+                result = run([repository.executable(), 'checkout', file], cwd=repository.root_path)
+                if result.returncode:
+                    sys.stderr.write('Failed to checkout file: {}'.format(file))
+                    run([repository.executable(), 'revert', '--abort'], cwd=repository.root_path)
+                    return 1
+
+        result = run([repository.executable(), 'revert', '--continue', '--no-edit'], cwd=repository.root_path)
+        if result.returncode:
+            run([repository.executable(), 'revert', '--abort'], cwd=repository.root_path)
+            sys.stderr.write('Failed revert commits')
+            return 1
+        log.info('Reverted {}'.format(','.join(commit_hashes)))
+        return 0
+    
+    @classmethod
+    def add_comment_to_reverted_commit_bug_tracker(cls, repository, args):
```

Comment:
Given that `create_pull_request` also makes a comment, I wonder if we should really provide a `.format`able string to `create_pull_request`, otherwise we'd end up with 2 comments referring to the same pull request